File: resources/expected/ros2/python-building_control_gen_mixed_inverted_topics/lax/src/building_control_py_pkg/building_control_py_pkg/base_code/tcp_tempControl_base_src.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from queue import Queue
from building_control_py_pkg.user_code.tcp_tempControl_src import *
from rclpy.callback_groups import ReentrantCallbackGroup
from building_control_py_pkg_interfaces.msg import Temperatureimpl
from building_control_py_pkg_interfaces.msg import FanAck
from building_control_py_pkg_interfaces.msg import SetPointimpl
from building_control_py_pkg_interfaces.msg import FanCmd
from building_control_py_pkg_interfaces.msg import Empty

#========================================================
# Re-running Codegen will overwrite changes to this file
#========================================================

class tcp_tempControl_base(Node):
    def __init__(self):
        super().__init__("tcp_tempControl")

        self.cb_group_ = ReentrantCallbackGroup()

        # Setting up connections
        self.tcp_tempControl_currentTemp_subscription_ = self.create_subscription(
            Temperatureimpl,
            "tcp_tempSensor_currentTemp",
            self.handle_currentTemp,
            1,
            callback_group=self.cb_group_)

        self.tcp_tempControl_fanAck_subscription_ = self.create_subscription(
            FanAck,
            "tcp_fan_fanAck",
            self.handle_fanAck,
            1,
            callback_group=self.cb_group_)

        self.tcp_tempControl_setPoint_subscription_ = self.create_subscription(
            SetPointimpl,
            "tcp_tempControl_setPoint",
            self.handle_setPoint,
            1,
            callback_group=self.cb_group_)

        self.tcp_tempControl_tempChanged_subscription_ = self.create_subscription(
            Empty,
            "tcp_tempSensor_tempChanged",
            self.event_handle_tempChanged,
            1,
            callback_group=self.cb_group_)

        self.tcp_tempControl_fanCmd_publisher_ = self.create_publisher(
            FanCmd,
            "tcp_tempControl_fanCmd",
            1)

    #=================================================
    #  C o m m u n i c a t i o n
    #=================================================

    def handle_currentTemp(self, msg):
        typedMsg = Temperatureimpl()
        typedMsg.data = msg
        self.currentTemp_msg_holder = typedMsg

    def get_currentTemp(self):
        return self.currentTemp_msg_holder

    def event_handle_tempChanged(self, msg):
         handle_tempChanged()

    def put_fanCmd(self, msg):
        typedMsg = FanCmd()
        typedMsg.data = msg
        self.tcp_tempControl_fanCmd_publisher_.publish(typedMsg)


    #=================================================
    #  C o m p u t e    E n t r y    P o i n t
    #=================================================
    def handle_fanAck(self, msg):
        raise NotImplementedError("Subclasses must implement this method")

    def handle_setPoint(self, msg):
        raise NotImplementedError("Subclasses must implement this method")

    def handle_tempChanged(self):
        raise NotImplementedError("Subclasses must implement this method")

File: resources/expected/ros2/building_control_gen_mixed_inverted_topics/lax/src/building_control_cpp_pkg/src/base_code/tcp_tempControl_base_src.cpp
#include "building_control_cpp_pkg/base_headers/tcp_tempControl_base_src.hpp"

//========================================================
// Re-running Codegen will overwrite changes to this file
//========================================================

tcp_tempControl_base::tcp_tempControl_base() : Node("tcp_tempControl")
{
    cb_group_ = this->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
    subscription_options_.callback_group = cb_group_;

    // Setting up connections
    tcp_tempControl_currentTemp_subscription_ = this->create_subscription<building_control_cpp_pkg_interfaces::msg::Temperatureimpl>(
        "tcp_tempSensor_currentTemp",
        1,
        std::bind(&tcp_tempControl_base::handle_currentTemp, this, std::placeholders::_1), subscription_options_);

    tcp_tempControl_fanAck_subscription_ = this->create_subscription<building_control_cpp_pkg_interfaces::msg::FanAck>(
        "tcp_fan_fanAck",
        1,
        std::bind(&tcp_tempControl_base::handle_fanAck, this, std::placeholders::_1), subscription_options_);

    tcp_tempControl_setPoint_subscription_ = this->create_subscription<building_control_cpp_pkg_interfaces::msg::SetPointimpl>(
        "tcp_tempControl_setPoint",
        1,
        std::bind(&tcp_tempControl_base::handle_setPoint, this, std::placeholders::_1), subscription_options_);

    tcp_tempControl_tempChanged_subscription_ = this->create_subscription<building_control_cpp_pkg_interfaces::msg::Empty>(
        "tcp_tempSensor_tempChanged",
        1,
        std::bind(&tcp_tempControl_base::event_handle_tempChanged, this, std::placeholders::_1), subscription_options_);

    tcp_tempControl_fanCmd_publisher_ = this->create_publisher<building_control_cpp_pkg_interfaces::msg::FanCmd>(
        "tcp_tempControl_fanCmd",
        1);

}

void tcp_tempControl_base::init_currentTemp(building_control_cpp_pkg_interfaces::msg::Temperatureimpl val) {
    currentTemp_msg_holder = std::make_shared<building_control_cpp_pkg_interfaces::msg::Temperatureimpl>(val);
}

//=================================================
//  C o m m u n i c a t i o n
//=================================================

void tcp_tempControl_base::handle_currentTemp(const building_control_cpp_pkg_interfaces::msg::Temperatureimpl::SharedPtr msg)
{
    currentTemp_msg_holder = msg;
}

building_control_cpp_pkg_interfaces::msg::Temperatureimpl::SharedPtr tcp_tempControl_base::get_currentTemp() {
    return currentTemp_msg_holder;
}

void tcp_tempControl_base::event_handle_tempChanged(const building_control_cpp_pkg_interfaces::msg::Empty::SharedPtr msg)
{
    (void)msg;
    handle_tempChanged();
}

void tcp_tempControl_base::put_fanCmd(building_control_cpp_pkg_interfaces::msg::FanCmd msg)
{
    tcp_tempControl_fanCmd_publisher_->publish(msg);
}

